<launch>
  <rosparam param="/use_sim_time">true</rosparam>
  <node respawn="true" name="map_server" pkg="map_server" type="map_server" args="/home/husarion/ros_workspace/src/ksce_demo/maps/demo_field1.yaml" machine="rosbot"/>
  <node name="amcl" pkg="amcl" type="amcl"  machine="rosbot">
    <param name="odom_alpha1" type="double" value="7" />
    <param name="odom_alpha2" type="double" value="7" />
    <param name="odom_alpha3" type="double" value="3.5" />
    <param name="odom_alpha4" type="double" value="3.5" />
    <param name="laser_max_beams" type="int" value="150" />
    <param name="min_particles" type="int" value="300" />
    <param name="max_particles" type="int" value="500" />
    <param name="update_min_d" type="double" value="0.05" />
    <param name="update_min_a" type="double" value="0.08" />
  </node>
  <node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode" output="screen" machine="rosbot">
    <param name="serial_port"         type="string" value="/dev/ttyUSB0"/>
    <param name="serial_baudrate"     type="int"    value="115200"/>
    <param name="frame_id"            type="string" value="laser"/>
    <param name="inverted"            type="bool"   value="false"/>
    <param name="angle_compensate"    type="bool"   value="true"/>
  </node>
  <node pkg="rosbot_webui" type="serial_bridge.sh" name="bridge" output="screen" machine="rosbot"/>

  <node pkg="tf" type="static_transform_publisher" name="ROSbot2_laser" args="0 0 0 3.14 0 0 base_link laser 10" machine="rosbot"/>

  <node pkg="tf" type="static_transform_publisher" name="ROSbot2_camera" args="-0.04 0 0.15 0 0 0 base_link camera_link 10" machine="rosbot"/>

  <node pkg="rosbot_webui" type="pose_to_tf_transform" name="pose_to_tf_transform" required="true" machine="rosbot"/>

  <node pkg="rosbot_webui" type="tf_to_pose" name="tf_to_pose" required="true" output="screen" machine="rosbot"/>

  <node pkg="husarion_ros" type="depth_clipping_node" name="depth_clipping_node" machine="rosbot"/>

  <node pkg="move_base" type="move_base" name="move_base" output="screen" machine="rosbot">
      <param name="recovery_behavior_enabled" value="true"/>
      <param name="controller_frequency" value="10.0"/>
      <rosparam file="$(find rosbot_webui)/conf/costmap_common_params.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find rosbot_webui)/conf/costmap_common_params.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find rosbot_webui)/conf/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find rosbot_webui)/conf/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find rosbot_webui)/conf/trajectory_planner.yaml" command="load" />
  </node>

  <node pkg="frontier_exploration" type="explore_client" name="explore_client" output="screen" machine="rosbot"/>

  <node pkg="frontier_exploration" type="explore_server" name="explore_server" output="screen" machine="rosbot">
      <param name="frequency" type="double" value="1.0"/>
      <param name="goal_aliasing" type="double" value="0.1"/>
      <rosparam ns="explore_costmap" subst_value="true" file="$(find rosbot_webui)/conf/exploration.yaml" command="load" />
  </node>

  <node name="robot_cmd" pkg="fiware_ros_bridge" type="robot_cmd.py" output="screen" machine="rosbot">
    <rosparam command="delete" param=""/>
    <rosparam command="load" file="$(find ksce_demo)/config/mqtt_local.yaml"/>
    <rosparam command="load" file="$(find fiware_ros_bridge)/config/robot_cmd.yaml"/>
  </node>

  <node pkg="ksce_demo" type="command_receiver.py" name="command_receiver_node" output="screen" machine="rosbot">
    <rosparam command="delete" param=""/>
    <rosparam command="load" file="$(find ksce_demo)/config/config.yaml"/>
  </node>

</launch>
